gene2phenotype_project/gene2phenotype_app/views/user.py

```diff
@@ -172,3 +284,146 @@ class ManageUserView(generics.RetrieveUpdateAPIView):
     def get_object(self):
         """Retrieve and return authenticated user"""
         return self.request.user
+
+class ChangePasswordView(generics.GenericAPIView):
+    """
+        Change password view  - Authenticated View 
+    """    
+    serializer_class = ChangePasswordSerializer
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request, *args, **kwargs):
+        """
+            Post method for ChangePasswordView
+
+            Args:
+                request (_type_): DjangoHttp Request object
+
+            Returns:
+                _type_: user information
+        """        
+        serializer = self.serializer_class(data=request.data, context={'user':request.user})
```

Comment:
Before serializer is called, we need a check to make sure the logged in user is the same as the username defined in data. Otherwise we can try to change other's passwords.